src/state/routing/types.ts

```diff
@@ -105,3 +104,43 @@ export class InterfaceTrade<
     this.gasUseEstimateUSD = gasUseEstimateUSD
   }
 }
+
+export class InterfaceTrade extends ClassicTrade<Currency, Currency, TradeType> {}
+
+export enum QuoteState {
+  SUCCESS = 'Success',
+  INITIALIZED = 'Initialized',
+  NOT_FOUND = 'Not found',
+}
+
+export type QuoteResult =
+  | {
+      state: QuoteState.INITIALIZED | QuoteState.NOT_FOUND
+      data?: undefined
+    }
+  | {
+      state: QuoteState.SUCCESS
+      data: GetQuoteResult
+    }
+
+export type TradeResult =
+  | {
+      state: QuoteState.INITIALIZED | QuoteState.NOT_FOUND
+      trade?: undefined
+    }
+  | {
+      state: QuoteState.SUCCESS
+      trade: InterfaceTrade
+    }
+
+export enum PoolType {
+  V2Pool = 'v2-pool',
+  V3Pool = 'v3-pool',
+}
+
+// swap router API special cases these strings to represent native currencies
+// all chains have "ETH" as native currency symbol except for polygon
```

Comment:
Is this due to a recent swap router upgrade, or is native polygon trading broken in its current state?

Reply:
its because native polygon is MATIC since they are a JOKE OF A CHAIN

Reply:
it's not broken because we always pass in `.wrapped` into the quote request, which effectively gives you the same result for native tokens (because on-chain they need to be wrapped anyway). but for [redacted] it wouldn't work simialrly